### PR TITLE
Remove `InputTooltip` component

### PR DESCRIPTION
**Pull request by lrhacker**
Now that the `Tooltip` component supports disabled inputs, we can remove the specialized `InputTooltip` component that was used as a workaround for this use case.

Unfortunately, due to how our actual `Input` and `Checkbox` components are normally styled and used, we still had to use the existing `Button` method that `InputTooltip` did to get the checkboxes to appear normally. I will be making a separate issue to look into fixing up the styles of the other components to not break the checkbox rendering by default, even when the checkbox has no label (e.g. set a default checkbox size, so they will never be rendered smaller than the expected 13x13px shown in Figma).

## Test plan
Check that tooltips on these inputs still behave as expected.

## App preview:

- [Web](https://sg-web-lrh-remove-input-tooltip.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-amnprsvpvb.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by courier-new** [comment]
Yessss, I'm sooo excited that new Tooltip doesn't have this problem!! 🙌 I recognize that almost all the usages of the workaround `InputTooltip` component were in the Batch Changes product area, so I wanted to offer a review and some support! 😄 The code changes look fine to me, but it looks like this is causing some visual regressions when the child `<Input/>` is `type="checkbox"`, such as the "select all" checkbox for changesets. I didn't look too closely, but it's likely related to the element hierarchy change, since these elements are typically positioned in a container with `display: grid`. Happy to look into it more or do more testing, if it'd be helpful. 🙂

File: client/wildcard/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -52,28 +53,12 @@ export const Tooltip: React.FunctionComponent<TooltipProps> = ({
...
-    }
-
     // NOTE: We plan to consolidate this logic with our Popover component in the future, but chose Radix first to support short-term accessibility needs.
     // GitHub issue: https://github.com/sourcegraph/sourcegraph/issues/36080
-    return (
-        <TooltipPrimitive.Root delayDuration={0} defaultOpen={defaultOpen}>
+    const tooltipRender = (trigger: React.ReactElement): JSX.Element => (
+        // A very small delay with the offset below helps prevent the tooltip from immediately closing when it gets triggered in the
+        // exact spot the arrow is overlapping the content
+        <TooltipPrimitive.Root delayDuration={100} defaultOpen={defaultOpen}>
```
Comment (jasongornall) [inline client/wildcard/src/components/Tooltip/Tooltip.tsx:61]: does this mean we don't need the alignOffset?

**Review by jasongornall — APPROVED** [review]

File: client/wildcard/src/components/Tooltip/Tooltip.tsx (hunk 1 above)
Comment (lrhacker) [reply client/wildcard/src/components/Tooltip/Tooltip.tsx:61]: The two together seemed to provide a good combination, but just the delay might be fine. I didn't spend much time comparing the two, just an initial test, so I'm fine removing offset now.

Comment (jasongornall) [reply client/wildcard/src/components/Tooltip/Tooltip.tsx:61]: yeah up to you whatever works 👍

**Comment by courier-new** [comment]
Looking a lot better, but it looks like the alignment is sometimes still off when the checkbox is disabled. 😞

image

It also looks a little smooshed, when they're all enabled?

image

(Sorry for the trouble. 🙇‍♀️)

**Comment by lrhacker** [comment]
I've decided to close this in favor of #38822 - the main priority right now is to remove `data-tooltip` uses, and I'll make a separate issue to track our style issues with using standalone checkboxes with tooltips for these batch changes grids.

**Pull request closed without merging**